compiler/rustc_monomorphize/src/collector/abi_check.rs

```diff
@@ -0,0 +1,119 @@
+use rustc_abi::Abi;
+use rustc_middle::ty::{self, Instance, InstanceKind, Ty, TyCtxt};
+use rustc_span::{def_id::DefId, Span, Symbol};
+use rustc_target::abi::call::{FnAbi, PassMode};
+
+use crate::errors::{AbiErrorDisabledVectorTypeCall, AbiErrorDisabledVectorTypeDef};
+
+const SSE_FEATURES: &'static [&'static str] = &["sse", "sse2", "ssse3", "sse3", "sse4.1", "sse4.2"];
```

Comment:
That's indeed what I did before, but it does not work. The feature check does not go through LLVM, and I did not find a way to make it do so...

Reply:
Please don't resolve comment when they haven't been resolved. :) For the final form of this PR we need to find something better than a random list of features burried deep inside rustc_monomorphize.

Reply:
Perhaps this list is unnecessary now after #128221?

Reply:
We still need to add the information about which target is needed for which vector size, but yeah the implications from https://github.com/rust-lang/rust/pull/128221 should be used here (or maybe they are already precomputed and can be used implicitly).

`compiler/rustc_target/src/target_features.rs` is probably also a better place for the vector-width tables, to keep it all contained in the same file.

Reply:
I got the impression this table needed to exist because you needed to know _every_ feature that uses e.g. ymm registers, but now there is the guarantee that if they are present, you will get the `avx` feature enabled.  So I think the table can be the simple `sse`, `avx`, `avx512f` table you originally suggested.

Reply:
I think I agree with your assessment - will try that out next week :-)